File: app/src/main/java/br/com/brqtest/model/Cliente.java
package br.com.brqtest.model;

import br.com.brqtest.R;
import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

import java.util.Date;

@DatabaseTable(tableName = "cliente")
public class Cliente extends EntidadeDominio {

    @DatabaseField(generatedId = true)
    private Integer id;
    @DatabaseField
    private String nameFull;
    @DatabaseField
    private String cpf;
    @DatabaseField(foreign = true)
    private Endereco endereco;
    @DatabaseField
    private Date dataDeNascimento;



    public Cliente(Integer id){
        this();
        setId(id);
    }

    public Cliente(){
        setEndereco(new Endereco());
    }

    public String getNameFull() {
        return nameFull;
    }

    public void setNameFull(String nameFull) {
        this.nameFull = nameFull;
    }

    public String getCpf() {
        return cpf;
    }

    public void setCpf(String cpf) {
        this.cpf = cpf;
    }

    public Endereco getEndereco() {
        return endereco;
    }

    public void setEndereco(Endereco endereco) {
        this.endereco = endereco;
    }

    public Date getDataDeNascimento() {
        return dataDeNascimento;
    }

    public void setDataDeNascimento(Date dataDeNascimento) {
        this.dataDeNascimento = dataDeNascimento;
    }

    @Override
    public String toString() {
        return "Cliente{" +
                "nameFull='" + nameFull + '\'' +
                ", cpf='" + cpf + '\'' +
                ", endereco=" + endereco +
                ", dataDeNascimento=" + dataDeNascimento +
                '}';
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }
}

File: app/src/main/java/br/com/brqtest/modelkotlin/Cliente.kt
package br.com.brqtest.modelkotlin

import com.j256.ormlite.field.DatabaseField
import com.j256.ormlite.table.DatabaseTable
import java.util.*

@DatabaseTable(tableName = "cliente")
class Cliente() : EntidadeDominio() {

    @DatabaseField(generatedId = true)
    var id: Int? = null
    @DatabaseField
    var nameFull: String? = null
    @DatabaseField
    var cpf: String? = null
    @DatabaseField(foreign = true, foreignAutoRefresh = true)
    var endereco: Endereco? = null
    @DatabaseField
    var dataDeNascimento: Date? = null



    init {
        endereco = Endereco()
    }

    override fun toString(): String {
        return "Cliente{" +
                "nameFull='" + nameFull + '\''.toString() +
                ", cpf='" + cpf + '\''.toString() +
                ", endereco=" + endereco +
                ", dataDeNascimento=" + dataDeNascimento +
                '}'.toString()
    }
}
